flow/project.py

```diff
@@ -2054,13 +2043,103 @@ def _get_cached_status(self):
         except KeyError:
             return {}
 
-    def _get_operations_status(self, jobs, cached_status):
-        """Return a dict with information about job-operations for this aggregate.
+    def _generate_selected_aggregate_groups(
```

Comment:
This is the most important part of this PR. We need this method to have a fast way to execute the iteration for all the cases below:
- `selected_aggregates` is `None`, a list of aggregates, an `_AggregateCursor`, or any other common way to express an iterable of aggregates
- `selected_groups` is `None` or a set of groups (e.g. groups representing singleton operations)

Reply:
This function is good to have, I agree that we need to cover these cases. Although its internal, and while it probably gets decent run from the various uses, I would like to see some direct tests of this method. It's complex enough, and right now if something goes wrong elsewhere we'd have to trace back pretty far to identify failure cases.

Reply:
I agree. I will add tests.